src/main/java/fi/otavanopisto/kuntaapi/server/images/ScaledImageStore.java

```diff
@@ -0,0 +1,138 @@
+import com.fasterxml.jackson.dataformat.smile.SmileFactory;
+
+import fi.otavanopisto.kuntaapi.server.id.AttachmentId;
+import fi.otavanopisto.kuntaapi.server.integrations.AttachmentData;
+
+/**
+ * Image store for scaled images
+ * 
+ * @author Antti Leppä
+ */
+@ApplicationScoped
+public class ScaledImageStore {
+ 
+  @Inject
+  private Logger logger;
+  
+  /**
+   * Returns scaled image data
+   * 
+   * @param kuntaApiAttachmentId attachmentId
+   * @param size size
+   * @return scaled image data or null if not found
+   */
+  public AttachmentData getStoredImage(AttachmentId kuntaApiAttachmentId, Integer size) {
+    File imageFile = getImageFile(kuntaApiAttachmentId, size);
+    if (imageFile.exists()) {
+      ObjectMapper objectMapper = getObjectMapper();
+      try {
+        return objectMapper.readValue(imageFile, AttachmentData.class);
+      } catch (IOException e) {
+        logger.log(Level.WARNING, String.format("Failed to read stored scaled image %s", imageFile.getName()), e);
+        if (!imageFile.delete()) {
+          logger.log(Level.WARNING, String.format("Failed to delete stored scaled image %s", imageFile.getName()), e);  
+        }
+      }
+    }
+    
+    return null; 
+  }
+  
+  /**
+   * Stores scaled image
+   * 
+   * @param kuntaApiAttachmentId attachmentId
+   * @param size size
+   * @param imageData image data
+   * @return image data
+   */
+  public AttachmentData storeScaledImage(AttachmentId kuntaApiAttachmentId, Integer size, AttachmentData imageData) {
+    File imageFile = getImageFile(kuntaApiAttachmentId, size);
+    
+    ObjectMapper objectMapper = getObjectMapper();
+    try {
+      objectMapper.writeValue(imageFile, imageData);
+    } catch (IOException e) {
+      logger.log(Level.WARNING, String.format("Failed to read stored scaled image %s", imageFile.getName()), e);
+      if (imageFile.exists() && !imageFile.delete()) {
+        logger.log(Level.WARNING, String.format("Failed to delete stored scaled image %s", imageFile.getName()), e);  
+      }
+    }
+    
+    return imageData;
+  }
+  
+  /**
+   * Lists all stored image sizes
+   * 
+   * @param kuntaApiAttachmentId attachmentId
+   * @return all stored image sizes
+   */
+  public List<Integer> listStoredImageSizes(AttachmentId kuntaApiAttachmentId) {
+    File[] storedFiles = listStoredImages(kuntaApiAttachmentId);
+
+    List<Integer> result = new ArrayList<>(storedFiles.length);
+    
+    for (File storedFile : storedFiles) {
+      result.add(Integer.parseInt(StringUtils.substringBeforeLast(storedFile.getName(), "-")));
+    }
+    
+    return result;
+  }
+  
+  /**
+   * Removes all stored image files
+   * 
+   * @param kuntaApiAttachmentId attachmentId
+   */
+  public void purgeStoredImages(AttachmentId kuntaApiAttachmentId) {
+    File[] storedImages = listStoredImages(kuntaApiAttachmentId);
+    for (File storedImage : storedImages) {
+      if (!storedImage.delete()) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Use "java.nio.Files#delete" here for better messages on error conditions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4042)